worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java

```diff
@@ -961,10 +961,15 @@ public int center(Region region, Pattern pattern) throws MaxChangedBlocksExcepti
         checkNotNull(pattern);
 
         Vector center = region.getCenter();
+        Vector vec1 = new Vector((int) center.getX(), (int) center.getY(), (int) center.getZ());
+        Vector vec2 = center.toBlockVector();
+        boolean condX = (region.getWidth() % 2) == 0;
+        boolean condZ = (region.getLength() % 2) == 0;
+        Vector vec3 = new Vector(vec2.getX() + (condX ? 0.5*Math.signum(vec2.getX()) : 0), vec2.getY(), vec2.getZ() + (condZ ? 0.5*Math.signum(vec2.getZ()) : 0));
```

Comment:
also if this is getting fixed again, can we do it correctly this time instead of relying on coordinate roundoff. 
why are we relying on a +/- 0.5 to be rounded to the correct side when we could just supply the right value to begin with.
this line is entirely unreadable and obfuscates the fact that we're relying on [these lines](https://github.com/sk89q/WorldEdit/blob/03f8843c2021425e5279aa82f9be6177d1802eed/worldedit-core/src/main/java/com/sk89q/worldedit/regions/CuboidRegion.java#L349-L351) to do the inference for us - this math isn't actually determining the center region on its own, it's just determining the correct "half block" we should be on so it later gets rounded to the right side.